homeassistant/components/husqvarna_automower_ble/__init__.py

```diff
@@ -23,9 +23,10 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Husqvarna Autoconnect Bluetooth from a config entry."""
     address = entry.data[CONF_ADDRESS]
+    pin = int(entry.data[CONF_PIN])
```

Comment:
Shouldn't we have a `if CONF_PIN not in entry.data: raise ConfigEntryAuthFailed` so it will just prompt users to add their stuff?

Reply:
I have added this, the problem is I can't figure out how to unit test the line now

Reply:
Creating a mock config entry without pin and then setting it up and then observing a flow has been created. `smartthings` has this added recently, check the test_init.py

Reply:
Thanks! Back to 100% test coverage